client/pages/fight.jsx

```diff
@@ -55,39 +50,55 @@ export default function Fight(props) {
     setTimeout(() => { setShowCountdown(false); }, 1000);
   }
 
-  function damage(player) {
-    const hit = 20;
-    if (player === 'you') {
-      setYourHp(yourHp - hit);
-    } else if (player === 'opp') {
-      setOppHp(oppHp - hit);
-    }
-  }
-
   // get metaData
-  useEffect(() => {
-    setMetaData(dummyMeta);
-    setYourId(dummyMeta.hostId);
-    setYourUsername(dummyMeta.hostUsername);
-    setOppUsername(dummyMeta.oppUsername);
+  useEffect(async () => {
+    const metaData = await fetch(`/api/game/${gameId}`)
+      .then(res => res.json())
+      .then(result => {
+        return result;
+      });
+    setMetaData(metaData);
+    if (!metaData.oppId) {
+      setYourId(metaData.hostId);
+      setYourUsername(metaData.hostName);
+      setOppUsername('Waiting for opponent...');
+    } else {
+      setYourId(metaData.oppId);
+      setYourUsername(metaData.oppName);
+      setOppUsername(metaData.hostName);
+      socket.current.emit('game joined', metaData);
+    }
   }, []);
 
   // socket connection
   useEffect(() => {
-    connectSocket(gameId, {
-      setPhrase,
-      damage
+    socket.current = io('/', { query: { gameId } });
+
+    socket.current.on('game joined', metaData => {
+      setMetaData(metaData);
+      setOppUsername(metaData.oppName);
+    });
+
+    socket.current.on('get random', phrase => {
+      setPhrase(phrase.content);
+    });
+
+    socket.current.on('finish phrase', damagedHp => {
+      setYourHp(damagedHp);
+      setPhrase('Getting phrase');
     });
     return () => {
-      disconnectSocket();
+      socket.current.disconnect();
     };
   }, []);
 
   // get new phrase
   useEffect(() => {
-    if (metaData) {
-      if (yourHp !== 0 && oppHp !== 0 && yourId === metaData.hostId) {
-        getRandom(gameId);
+    if (metaData && metaData.oppId) {
+      if (yourHp !== 0 && oppHp !== 0) {
+        if (yourId === metaData.hostId) {
+          socket.current.emit('get random', gameId);
+        }
```

Comment:
Can probably join all of these nested conditions together into one since we're ultimately performing one action